addons/web/static/src/legacy/js/views/basic/basic_model.js

```diff
@@ -4053,6 +4054,8 @@ var BasicModel = AbstractModel.extend({
      * @param {boolean} [params.static=false]
      * @param {string} [params.type='record'|'list']
      * @param {[type]} [params.value]
+     * @param {Object} [params.range] only for date/datetime fields : {[fieldName]: {from: string, to: string}}
+     *   where from (inclusive) and to (exclusive) are the date bounds of the datapoint under the respective format.
```

Comment:
docstring is a bit incomplete. As far as I understand, this only concerns datapoints representing groups, and when the group comes from a groupby on a date(time) field, a range is given, and corresponds to what you wrote.

Question: if you group on several levels, say first on a date field A and then on a datetime field B, and to make it easy you have a single leaf group (i.e. all records are in the same group). You'll get two group datapoints: one for the first level of grouping, containing the subgroup, and one for that subgroup. What will be the `range` value for those two groups? I would expect that for the first one, it only contains the range for A, and for the subgroup, the range for B. Is it the case? Or do you pass the full range object at every level?

Reply:
As of now in lazy mode (read_group), the ranges would be (I used 3 levels of grouping instead of 2):
read_group on : A > B > C
```
firstLevelGroupRange = {
  A: {from:..., to:...},
  B: false,
  C: false,
}
secondLevelGroupRange = {
  B: {from:..., to:...},
  C: false,
}
thirdLevelGroupRange = {
  C: {from:..., to:...},
}
```
* At each level, "non-consumed" GroupByFieldNames have their range at `false`
* At each next level, the range lose the "consumed" GroupByFieldNames